my/core/query_range.py

```diff
@@ -73,13 +73,28 @@ def parse_datetime_float(date_str: str) -> float:
         return ds_float
     try:
         # isoformat - default format when you call str() on datetime
+        # this also parses dates like '2020-01-01'
         return datetime.fromisoformat(ds).timestamp()
     except ValueError:
         pass
     try:
         return isoparse(ds).timestamp()
     except (AssertionError, ValueError):
-        raise QueryException(f"Was not able to parse {ds} into a datetime")
+        pass
+
+    try:
+        import dateparser # type: ignore[import]
+        # dateparser is a bit more lenient than the above, lets you type
+        # all sorts of dates as inputs
+        # https://github.com/scrapinghub/dateparser#how-to-use
+
+        res: Optional[datetime] = dateparser.parse(ds)
+        if res is not None:
+            return res.timestamp()
+    except ImportError:
```

Comment:
nit, but probably worth wrapping only import in the `except` statement, then the rest of the code can be in `else` block